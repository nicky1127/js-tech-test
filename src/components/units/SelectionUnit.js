import React from "react";
import PropTypes from "prop-types";
import clsx from "clsx";
import { connect } from "react-redux";
import { makeStyles } from "@material-ui/core/styles";
import { Grid, Box } from "@material-ui/core";
import { ExpandMore, Clear } from "@material-ui/icons";
import ScoreTag from "components/units/ScoreTag";
import { removeSelection } from "redux/_actions";
import labels from "constants/labels";

const constants = labels.SelectionUnit;

const useStyles = makeStyles(theme => ({
  selectionUnit: {
    padding: "8px",
    backgroundColor: "rgb(239,244,252)",
    paddingBottom: "15px",
    border: "1px solid rgb(179,191,211)"
  },
  selectionHeader: { fontWeight: "600", fontSize: "15px" },
  event: {
    padding: "10px 0",
    fontSize: "12px",
    width: "80%"
  },
  crossIcon: {
    display: "flex",
    alignItems: "flex-start",
    justifuContent: "flex-end"
  }
}));

export const SelectionUnit = props => {
  const classes = useStyles();
  const { isOddsDecimal, outcome, removeSelection } = props;

  const price =
    outcome &&
    outcome.price &&
    (isOddsDecimal
      ? outcome.price.decimal
      : `${outcome.price.num}/${outcome.price.den}`);

  const onClickBtn = () => {
    removeSelection(outcome.outcomeId);
  };

  return (
    <Box className={classes.selectionUnit} data-testid="selectionUnit">
      <Grid container spacing={0}>
        <Grid item sm={10} className={classes.selectionHeader}>
          {`${outcome.name} @ `}
          <span style={{ color: "rgb(217,34,49)", fontWeight: "400" }}>
            {price}
          </span>
        </Grid>
        <Grid item sm={2} className={classes.crossIcon}>
          <Clear onClick={onClickBtn} />
        </Grid>
        <Grid item sm={12} className={classes.event}>
          {`${outcome.eventName} - ${outcome.marketName}`}
        </Grid>
      </Grid>
    </Box>
  );
};

SelectionUnit.defaultProps = { outcome: {} };

SelectionUnit.propTypes = {};

const mapStateToProps = state => {
  const { isOddsDecimal } = state;
  return { isOddsDecimal };
};

const ConnectedSelectionUnit = connect(mapStateToProps, {
  removeSelection
})(SelectionUnit);

export default ConnectedSelectionUnit;
