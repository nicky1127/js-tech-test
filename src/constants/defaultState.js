const state = {
  liveEvents: [],
  event: {},
  markets: {},
  isOddsDecimal: false,
  loading: false,
  error: {},
  selections: []
};

export default state;
