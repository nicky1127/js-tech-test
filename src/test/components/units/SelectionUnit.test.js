import React from "react";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import SelectionUnit from "components/units/SelectionUnit";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import defaultState from "constants/defaultState";

beforeEach(() => {
  jest.clearAllMocks();
  cleanup();
});

const props = {};

const state = {
  ...defaultState
};
const mockStore = configureMockStore();
const store = mockStore(state);

describe("SelectionUnit", () => {
  it("Snapshot matchs", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <SelectionUnit {...props} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("SelectionUnit should render one [data-testid=SelectionUnit]", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <SelectionUnit {...props} />
      </Provider>
    );
    const container = getByTestId("selectionUnit");
    expect(container).toBeInTheDocument();
  });
});
