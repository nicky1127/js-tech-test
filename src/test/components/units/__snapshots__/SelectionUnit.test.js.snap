// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectionUnit Snapshot matchs 1`] = `
<div
  className="MuiBox-root MuiBox-root-5 makeStyles-selectionUnit-1"
  data-testid="selectionUnit"
>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root makeStyles-selectionHeader-2 MuiGrid-item MuiGrid-grid-sm-10"
    >
      undefined @ 
      <span
        style={
          Object {
            "color": "rgb(217,34,49)",
            "fontWeight": "400",
          }
        }
      />
    </div>
    <div
      className="MuiGrid-root makeStyles-crossIcon-4 MuiGrid-item MuiGrid-grid-sm-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </div>
    <div
      className="MuiGrid-root makeStyles-event-3 MuiGrid-item MuiGrid-grid-sm-12"
    >
      undefined - undefined
    </div>
  </div>
</div>
`;
